src/client/datascience/types.ts

```diff
@@ -433,6 +433,7 @@ export interface IInteractiveWindow extends IInteractiveBase {
     readonly readyPromise: Promise<void>;
     readonly kernelPromise: Promise<IKernel | undefined>;
     closed: Event<void>;
+    ensureKernelReadyPromise(): Promise<IKernel>;
```

Comment:
I don't think this is necessary as you can do this in the 'readyPromise' getter.